python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1753 @@
+                    alt_in_position
+                    and az_in_position
+                    and na1_in_position
+                    and na2_in_position
+                    and dom_in_position
+                )
+            elif atdome:
+                self.log.info(f"[Dome] delta Az = {dom_az_dif:+08.3f}")
+                in_position = dom_in_position
+            elif atmcs:
+                self.log.info(
+                    f"[Telescope] delta Alt = {alt_dif:+08.3f} | delta Az= {az_dif:+08.3f} "
+                    f"delta N1 = {nasm1_dif:+08.3f} delta N2 = {nasm2_dif:+08.3f} "
+                )
+                in_position = (
+                    alt_in_position
+                    and az_in_position
+                    and na1_in_position
+                    and na2_in_position
+                )
+            else:
+                break
+
+            await asyncio.sleep(1.0)
+
+        if in_position:
+            self.log.debug("Axes in position.")
+        else:
+            self.log.warning("Nothing to check.")
+
+    async def check_target_status(self):
+        """Check the targeting status of the atmcs.
+        """
+        while True:
+            in_position = await self.atmcs.evt_allAxesInPosition.next(flush=False)
+            self.log.debug(f"Got {in_position.inPosition}")
+            if in_position.inPosition is False:
+                raise RuntimeError(f"ATMCS is no longer tracking.")
+
+    async def focus_offset(self, offset):
+        """ Apply focus offset.
+
+        Returns after offset is aplied.
+
+        Parameters
+        ----------
+        offset : float
+            Offset in mm.
+
+        """
+        self.athexapod.evt_positionUpdate.flush()
+        await self.ataos.cmd_applyFocusOffset.set_start(offset=offset)
+
+        try:
+            await self.athexapod.evt_positionUpdate.next(
+                flush=False, timeout=self.long_timeout
+            )
+        except asyncio.TimeoutError:
+            pass
+
+    async def add_point_data(self):
+        """ Add current position to a point file. If a file is open it will
+        append to that file. If no file is opened it will open a new one.
+
+        """
+
+        state = await self.get_state("atptg")
+        if state != salobj.State.ENABLED:
+            raise RuntimeError(
+                f"ATPtg in {state!r}. Expected {salobj.State.ENABLED!r}."
+            )
+
+        try:
+            await self.atptg.cmd_pointAddData.start()
+        except salobj.AckError:
+            self.log.debug("Opening new pointing file.")
+            await self.atptg.cmd_pointNewFile.start()
+            await self.atptg.cmd_pointAddData.start()
+
+    @property
+    def valid_use_cases(self):
+        """Returns valid usages.
+
+        When subclassing, overwrite this method to return the proper enum.
+
+        Returns
+        -------
+        usages: enum
+
+        """
+        return ATCSUsages
+
+    @property
+    def usages(self):
```

Comment:
Please document what this returns.

Also please consider using a custom struct-like class for this purpose (e.g. `Usage`). Advantages:
* It is a natural place to document to document what a Usage is (I have no idea).
* You can add methods that manage the data
* It reduces the risk of typos or code evolution resulting in extra unwanted fields.